### ISSUE TITLE
Setup changes VBA Settings

### ISSUE THREAD

**Issue by SchoepkeD** [rubberduck-vba/Rubberduck#4165]
Hello,
first off, many thanks for Rubberduck! I was waiting for something like this to come along for a long time now!

But there is a problem. The add-in parses everything in normal.dot. Then comes up with a bunch of errors (e.g. variable not declared) that were no problem until now. Easy fix though.

But also, some of my add-ins are getting deactivated, I can't run them, and I can't find the reason for the life of me. It says something with ByRef expected or something. I need that add-in, so I de-install RD, but after restarting word, the problems remain. I resort to firing up Acronis and restoring the whole partition C: as a last panic resort.

This time loss could have been easily avoided if RD told me EXACTLY what it changes, if at all (it shouldn't without asking first. Common courtesy). *I* want to have the last word, and not some program automatically garble up my whole installation without so much as a whoopsie-daisy!

I have installed RD on Windows 7.1 x64 German, using it in Office 2003.

Please furnish options for non-intrusive setup, a roll-back option, a log file that is automatically opened to tell me what tf it has done to my Installation! This is just rude!

Best,
Dieter

**Comment by bclothier** [comment]
While I agree this is very undesirable and I would indeed would be annoyed as you are if it messed up things like that.

However, there are few things we need to help us understand what exactly went wrong. We do try to make our install non-intrusive as much as we can. Unless I'm mistaken, this is a first report where another add-in got affected, and this is now how RD installation should work.

What add-ins got deactivated here? Was it a Word add-in or a VBA add-in? 

If it's a Word add-in, does it modify the normal.dotx? 

Did the problem start immediately as soon as you installed Rubberduck, or was it only after you applied some fixes? If latter, did you apply fixes on the normal.dotx's codebase or was it only within your VBA project? 

The more details you provide, the better we can help.

**Comment by rubberduck203** [comment]
I’m not really sure what to say here. We’re very clear about requiring admin privileges for installation in the docs linked in the readme and how to perform a per-user install if that doesn’t suit you. The per user install instructions refer to the code that modifies your registry. We’re not hiding anything. Unfortunately, the only way to extend the VBE is to do so via implementing a COM interface and registering those COM components. 

I’m sorry you had a poor experience, but as with any other add-in, we can’t guarantee that RD won’t conflict with another add-in you’ve got installed. 

I’m equally sorry that I need to close this until you provide enough details to reproduce the issue you’ve seen. 

- What other add-ins do you have installed?
- Office 2003 is 32bit only, correct?
- What OS are you running?

**Comment by retailcoder** [comment]
Ok. I'll ignore the incredibly rude tone and ask one thing: you say "setup changes VBA settings" - **what** settings specifically? Because no, Rubberduck doesn't mess with your VBA settings, and is very transparent about everything it does.

This issue is conflating misunderstanding of static code analysis results, compile errors you (or some Rubberduck quickfix?) introduced in your add-ins' code, and other things about the installer being "intrusive", whatever that means.

Is there something we can do to help, or this was just venting?

**Comment by SchoepkeD** [comment]
bclothier,
it were two of my five hand-written add-ins. It's all in VBA and linked to from Extras, Vorlagen und Add-Ins (Templates and Add-ins?). Basically they're macro-libraries I want to switch off. It's Office 2003, so "dot", not "dotx"

rubberduck203,
what do the privileges have to do with this? It installed, and it anihilated, so privileges were sufficient. I wrote answers to all your questions above, so just read the WHOLE thing pls. Office 2k3 is 32bit only.

And Sheriff retailshmoder,
if you don't have anything to say at all, shut up. This aint facebook.

Best,
Dieter

**Comment by Vogel612** [comment]
If you're not capable of interacting with the maintainers of the addin on a professional level, you're not worthy of our time. If you have an actual issue to report that's more than a hidden series of insults, you're welcome to do so in a separate issue.

This issue is now closed. Do note that Rubberduck does **not** change your VBA / VBE settings in any way without asking beforehand. Do also note that Rubberduck inspections are not compile errors and that you should understand what Rubberduck will do before applying any of it's suggestions.

Thank you for understanding
